tests/unittests/test_datasource/test_gce.py

```diff
@@ -360,5 +360,27 @@ def test_publish_host_keys(self, m_readurl):
         self.ds.publish_host_keys(hostkeys)
         m_readurl.assert_has_calls(readurl_expected_calls, any_order=True)
 
+    @mock.patch(
+        "cloudinit.sources.DataSourceGCE.EphemeralDHCPv4",
+        autospec=True,
+    )
+    @mock.patch(
+        "cloudinit.sources.DataSourceGCE.DataSourceGCELocal.fallback_interface"
+    )
+    def test_local_datasource_uses_ephemeral_dhcp(self, _m_fallback, m_dhcp):
+        ds = DataSourceGCE.DataSourceGCELocal(
+            sys_cfg={}, distro=None, paths=None
+        )
+        ds._get_data()
+        m_dhcp.assert_called_once()
```

Comment:
DNE, you probably meant assert_called_once_with. But we can instead just check call_count
```suggestion
        self.assertEqual(1, m_dhcp.call_count)
```

Reply:
Heh...so since I knew the death of 3.5 is imminent for cloud-init, and Brett showed me [a really cool package](https://pypi.org/project/darker/) for black formatting of only changed code that is 3.6+, I figured I can upgrade my local dev environment because am I really going to run into any python 3.5-only stuff in the next 3 weeks?

Apparently the answer is [yes](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once) :man_facepalming: